packages/samples/headless-ssr/src/app/generic/hooks/search-parameters.ts

```diff
@@ -0,0 +1,111 @@
+'use client';
+
+import {SearchParameters} from '@coveo/headless';
+import {
+  SearchParameterManager,
+  SearchParameterManagerState,
+} from '@coveo/headless/ssr';
+import {useCallback, useEffect, useMemo, useReducer, useState} from 'react';
+import {CoveoNextJsSearchParametersSerializer} from '../common/search-parameters-serializer';
+
+interface UseSyncSearchParametersProps {
```

Comment:
Can't we use the `useSearchParams()` nextjs hook?

Reply:
Not for this one. The purpose of this hook is to sync the search parameters bi-directionally with the controller. The purpose of `useSearchParams` is to read the URL search parameters.

If what you meant to ask is whether `useUrlSearchParams` could be replaced with `useSearchParams`, I just tested it and that works, so I'll replace it. I don't remember why I didn't do it like that in the first place.

Reply:
Update: turns out that `useSearchParams` doesn't refresh when we push new history states, woops.